include/onnxruntime/core/framework/float16.h

```diff
@@ -17,53 +27,64 @@ struct MLFloat16 {
 
   float ToFloat() const;
 
-  operator float() const {
-    return ToFloat();
-  }
+  operator float() const { return ToFloat(); }
 };
 
-inline bool operator==(const MLFloat16& left, const MLFloat16& right) {
-  return left.val == right.val;
-}
+inline bool operator==(const MLFloat16& left, const MLFloat16& right) { return left.val == right.val; }
+inline bool operator!=(const MLFloat16& left, const MLFloat16& right) { return left.val != right.val; }
+inline bool operator<(const MLFloat16& left, const MLFloat16& right) { return left.val < right.val; }
 
-inline bool operator!=(const MLFloat16& left, const MLFloat16& right) {
-  return left.val != right.val;
-}
-
-inline bool operator<(const MLFloat16& left, const MLFloat16& right) {
-  return left.val < right.val;
-}
-
-//BFloat16
+// BFloat16
 struct BFloat16 {
   uint16_t val{0};
-  explicit BFloat16() = default;
-  explicit BFloat16(uint16_t v) : val(v) {}
-  explicit BFloat16(float v) {
+#if defined(USE_ROCM)
+  ORT_HOST_DEVICE BFloat16() = default;
```

Comment:
Out of curiosity, why is the line above specific to ROCM?

Reply:
I saw PyTorch does this way for all default constructors so I followed the same way. Maybe hipcc requires this? But I didn't find out any documentation to support this.

Reply:
OK, let's leave it as it is and will re-visit when supporting BF16 on AMD GPU.